osu.Game.Rulesets.Mania/Mods/ManiaModClassic.cs

```diff
@@ -1,11 +1,46 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using osu.Framework.Bindables;
+using osu.Game.Beatmaps;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Mania.Objects;
+using osu.Game.Rulesets.Mania.Scoring;
 using osu.Game.Rulesets.Mods;
 
 namespace osu.Game.Rulesets.Mania.Mods
 {
-    public class ManiaModClassic : ModClassic
+    public class ManiaModClassic : ModClassic, IApplicableToBeatmap
     {
+        [SettingSource("Use classic hit windows", "Changes timing windows to match previous versions of osu! exactly.")]
+        public Bindable<bool> ClassicHitWindows { get; } = new BindableBool(true);
+
+        public void ApplyToBeatmap(IBeatmap beatmap)
+        {
+            bool isConvert = beatmap.BeatmapInfo.Ruleset.ShortName != ManiaRuleset.SHORT_NAME;
```

Comment:
I'm slightly weirded out that this is using a different method of comparing rulesets than the one already established by [`ManiaBeatmapConverter`](https://github.com/ppy/osu/blob/e97c75708484030ce6b57080576ccfd368c5e31e/osu.Game.Rulesets.Mania/Beatmaps/ManiaBeatmapConverter.cs#L63), and equatable in question differs for [`APIRuleset`](https://github.com/ppy/osu/blob/e97c75708484030ce6b57080576ccfd368c5e31e/osu.Game/Online/API/Requests/Responses/APIBeatmap.cs#L173).

Maybe `beatmap.BeatmapInfo.Ruleset.Equals(new ManiaRuleset().RulesetInfo)`?